shared/teams/team/member/add-to-channels.tsx

```diff
@@ -186,96 +189,224 @@ const AddToChannels = (props: Props) => {
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const isAdmin = false
+  const inChannel = meta.membershipType === 'active'
+  const onEditChannel = () => {} // TODO: show another modal with a back button to here
+  const actionProps = {conversationIDKey: meta.conversationIDKey, teamID: meta.teamID}
+  const onAdministrate = () =>
+    dispatch(nav.safeNavigateAppendPayload({path: [{props: actionProps, selected: 'teamChannel'}]}))
+  const onDelete = () =>
+    // TODO: reload this list or at least hide this row
+    dispatch(nav.safeNavigateAppendPayload({path: [{props: actionProps, selected: 'teamDeleteChannel'}]}))
+  const onLeave = () =>
+    dispatch(
+      ChatGen.createLeaveConversation({conversationIDKey: meta.conversationIDKey, dontNavigateToInbox: true})
+    )
+  const onJoin = () => dispatch(ChatGen.createJoinConversation({conversationIDKey: meta.conversationIDKey}))
+
+  const menuItems = [
+    {onClick: onEditChannel, title: 'Edit channel'},
+    ...(isAdmin
+      ? [
+          {onClick: onAdministrate, title: 'Administrate'},
+          {danger: true, onClick: onDelete, title: 'Delete'},
+        ]
+      : []),
+  ]
+  const {popupAnchor, showingPopup, toggleShowingPopup, popup} = Kb.usePopup(getAttachmentRef => (
+    <Kb.FloatingMenu
+      attachTo={getAttachmentRef}
+      visible={showingPopup}
+      onHidden={toggleShowingPopup}
+      closeOnSelect={true}
+      items={menuItems}
+    />
+  ))
+  const [buttonMousedOver, setMouseover] = React.useState(false)
+  return (
+    <Kb.Box2 direction="horizontal" gap="xtiny" fullHeight={true}>
+      {popup}
+      {meta.channelname !== 'general' && (
+        // TODO: waiting state
+        // TODO: flip this to the other button after being clicked
+        <Kb.Button
+          type={buttonMousedOver && inChannel ? 'Default' : 'Success'}
+          mode={inChannel ? 'Secondary' : 'Primary'}
+          label={!inChannel ? 'Join' : buttonMousedOver ? 'Leave' : 'In'}
+          icon={buttonMousedOver || !inChannel ? undefined : 'iconfont-check'}
+          iconSizeType={buttonMousedOver || !inChannel ? undefined : 'Tiny'}
+          onMouseEnter={Styles.isMobile ? undefined : () => setMouseover(true)}
+          onMouseLeave={Styles.isMobile ? undefined : () => setMouseover(false)}
+          onMouseDown={
+            Styles.isMobile
+              ? undefined
+              : evt => {
+                  // using onMouseDown so we can prevent blurring the search filter
+                  evt.preventDefault()
+                  inChannel ? onLeave() : onJoin()
+                }
+          }
+          onClick={Styles.isMobile ? (inChannel ? onLeave : onJoin) : undefined}
+          small={true}
+          style={styles.joinLeaveButton}
+        />
+      )}
+      <Kb.Button
+        icon="iconfont-ellipsis"
+        iconColor={Styles.globalColors.black_50}
+        // TODO: this isn't working on desktop
```

Comment:
just switch to `onClick`

Reply:
tl;dr I think this is because of the order click events fire and the FloatingMenu's onhidden happens later.

Reply:
😢